daemon/endpoint.go

```diff
@@ -251,7 +251,7 @@ func (d *Daemon) createEndpoint(ctx context.Context, epTemplate *models.Endpoint
 		return d.errorDuringCreation(ep, fmt.Errorf("unable to insert endpoint into manager: %s", err))
 	}
 
-	ep.UpdateLabels(ctx, addLabels, infoLabels, true)
+	ep.UpdateLabels(ctx, addLabels, infoLabels, true, d.identityAllocator)
```

Comment:
Did you consider having these functions call via the owner rather than explicitly passing the identityAllocator everywhere? Is it clearer this way?

Reply:
I think it's clearer this way, personally. The `Owner` is a way of providing a way of hooking global state into packages without actually plumbing values down; it sometimes acts on global variables in packages that cannot be imported due to import loops. I think that having a smaller, more specific interface and providing that as a parameter allows for easier mocking (no need to implement all of the `Owner` functions just for identity allocation, for instance).

Reply:
Fixed in latest upload; the Endpoint now has a member which it calls to perform identity-related operations.